lib/go/thrift/simple_server.go

```diff
@@ -149,8 +151,10 @@ func (p *TSimpleServer) Serve() error {
 }
 
 func (p *TSimpleServer) Stop() error {
-	p.quit <- struct{}{}
-	p.serverTransport.Interrupt()
+	if atomic.CompareAndSwapInt64(&p.stopped, 0, 1) {
```

Comment:
I created a Jira item for your issue:

https://issues.apache.org/jira/browse/THRIFT-3855